graphql/resolvers.go

```diff
@@ -2853,6 +2872,111 @@ func (r *versionResolver) BuildVariants(ctx context.Context, v *restModel.APIVer
 }
 
+func (r *versionResolver) IsPatch(ctx context.Context, v *restModel.APIVersion) (bool, error) {
+	return evergreen.IsPatchRequester(*v.Requester), nil
+}
+
+func (r *versionResolver) Patch(ctx context.Context, v *restModel.APIVersion) (*restModel.APIPatch, error) {
+	if !evergreen.IsPatchRequester(*v.Requester) {
+		return nil, nil
+	}
+	apiPatch, err := r.sc.FindPatchById(*v.Id)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Couldn't find a patch with id: `%s` %s", *v.Id, err.Error()))
+	}
+	return apiPatch, nil
+}
+
+func (r *versionResolver) TaskCount(ctx context.Context, obj *restModel.APIVersion) (*int, error) {
+	taskCount, err := task.Count(task.ByVersion(*obj.Id))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting task count for version `%s`: %s", *obj.Id, err.Error()))
+	}
+	return &taskCount, nil
+}
+
+func (r *versionResolver) VersionTiming(ctx context.Context, obj *restModel.APIVersion) (*VersionTiming, error) {
+	tasks, err := task.FindAll(task.ByVersion(*obj.Id).WithFields(task.TimeTakenKey, task.StartTimeKey, task.FinishTimeKey, task.DisplayOnlyKey, task.ExecutionKey))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding tasks for version `%s` : %s", *obj.Id, err.Error()))
+	}
+	if tasks == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Could not find any tasks for version %s", *obj.Id))
+	}
+	timeTaken, makespan := task.GetTimeSpent(tasks)
+
+	// return nil if rounded timeTaken/makespan == 0s
+	t := timeTaken.Round(time.Second)
+	m := makespan.Round(time.Second)
+
+	var apiTimeTaken *restModel.APIDuration
+	var apiMakespan *restModel.APIDuration
+	if t.String() != "0s" {
+		apiDuration := restModel.NewAPIDuration(t)
+		apiTimeTaken = &apiDuration
+	}
+	if m.String() != "0s" {
+		apiDuration := restModel.NewAPIDuration(m)
+		apiMakespan = &apiDuration
+	}
+
+	return &VersionTiming{
+		TimeTaken: apiTimeTaken,
+		Makespan:  apiMakespan,
+	}, nil
+}
+
+func (r *versionResolver) BaseVersionID(ctx context.Context, obj *restModel.APIVersion) (*string, error) {
+	baseVersion, err := model.VersionFindOne(model.BaseVersionByProjectIdAndRevision(*obj.Project, *obj.Revision).WithFields(model.VersionIdentifierKey))
+	if baseVersion == nil || err != nil {
+		return nil, nil
+	}
+	return &baseVersion.Id, nil
+}
+
+func (r *versionResolver) Status(ctx context.Context, obj *restModel.APIVersion) (string, error) {
+	failedAndAbortedStatuses := append(evergreen.TaskFailureStatuses, evergreen.TaskAborted)
+	opts := data.TaskFilterOptions{
+		Statuses:        failedAndAbortedStatuses,
+		FieldsToProject: []string{task.DisplayStatusKey},
+	}
+	tasks, _, err := r.sc.FindTasksByVersion(*obj.Id, opts)
+	if err != nil {
+		return "", InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for version: %s ", err.Error()))
+	}
+	status, _ := evergreen.VersionStatusToPatchStatus(*obj.Status)
```

Comment:
If we have an unknown status that _is_ pretty bad; I think it's worth at least logging the status to splunk so that if we ever _do_ get a blank status we'll be able to look up what happened.